packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts

```diff
@@ -560,6 +595,17 @@ export class Metric implements IMetric {
     });
   }
 
+  /**
+   * Make a new Alarm for this metric using anomaly detection
+   *
+   * @param scope The scope in which to create the alarm
+   * @param id The ID of the alarm
+   * @param props Alarm creation properties for anomaly detection
+   */
+  public createAnomalyDetectionAlarm(scope: Construct, id: string, props: CreateAnomalyDetectionAlarmProps): Alarm {
+    return Metric.createAnomalyDetectionAlarmFromMetric(scope, id, props, this);
+  }
```

Comment:
Adding these helpers to `IMetric` feels unnecessary. I understand you're probably doing it because `createAlarm()` exists, but I regret that as well :). 

What about 2 static factory functions, and then we're doing:

```ts
Alarm.createAnomalyDetectionAlarm(originalMetric, { stdDevs: 1, ... })

Metric.createAnomalyDetectionBandMetric(originalMetric, { stdDevs: 1, ... })
```

And that's enough?